doc/sql.extensions/README.blob_util.md

```diff
@@ -0,0 +1,184 @@
+# `RDB$BLOB_UTIL` package (FB 5.0)
+
+Before Firebird 5 BLOBs could be created appending data using the concatenate (`||`) operator with another BLOB.
+
+However there is two big problems with this approach:
+- BLOBs are always created in database page space, even if that is not required (not going to a persistent table).
+- Every time a concatenation happens, a new BLOB is created, filling the database with garbage.
+
+`RDB$BLOB_UTIL` package now exists to fix that problems.
+
+It allows BLOB creation in temporary page space (as well in the persistent page space) and has a workflow so that a BLOB is created and could have its data incrementally appended before it's closed and be available as a BLOB value. This is much like the BLOB client API works, but now available in PSQL.
+
+## Function `NEW`
+
+`RDB$BLOB_UTIL.NEW` is used to create a new BLOB. It returns a handle (an integer bound to the transaction) that should be used with the others functions of the package.
+
+Input parameter:
+ - `SEGMENTED` type `BOOLEAN NOT NULL`
+ - `TEMP_STORAGE` type `BOOLEAN NOT NULL`
+
+Return type: `INTEGER NOT NULL`.
+
+## Function `OPEN_BLOB`
+
+`RDB$BLOB_UTIL.OPEN_BLOB` is used to open an existing BLOB for read. It returns a handle that should be used with the others functions of the package.
+
+Input parameter:
+ - `BLOB` type `BLOB NOT NULL`
+
+Return type: `INTEGER NOT NULL`.
+
+## Procedure `APPEND`
+
+`RDB$BLOB_UTIL.APPEND` is used to append chunks of data to a BLOB handle created with `RDB$BLOB_UTIL.NEW`.
+
+Input parameters:
+ - `HANDLE` type `INTEGER NOT NULL`
+ - `DATA` type `VARBINARY(32767) NOT NULL`
+
+## Function `READ`
+
+`RDB$BLOB_UTIL.READ` is used to read chunks of data of a BLOB handle opened with `RDB$BLOB_UTIL.OPEN_BLOB`. When the BLOB is fully read and there is no more data, it returns `NULL`.
+
+If `LENGTH` is passed with a positive number, it returns a VARBINARY with its maximum length.
+
+If `LENGTH` is `NULL` it returns just a segment of the BLOB with a maximum length of 32765.
+
```

Comment:
We could increase max VARCHAR to 64KB - 2. But also, is there an impediment to have max dsc_length of dtype_varying to 64KB?

As I understand, there should not be many places just reading and incrementing dsc_length, so dtype_cstring / dtype_varying does not need to have the constant size added to dsc_length.

It would simplify various places that substract and re-add that value.